### PR TITLE
fix MD5 length and compile error

### LINKED ISSUE #1041: MD5 and MD4 incorrect byteLength
### Version

0.1.7

### Platform

Linux 5.10.102.1-microsoft-standard-WSL2

### What steps will reproduce the bug?

Code to reproduce:

```typescript
import { MD4, MD5 } from "bun";

console.log(new MD5().update('Hola').digest('hex'));
console.log(new MD4().update('Hola').digest('hex'));
console.log(MD4, MD5);
```

### How often does it reproduce? Is there a required condition?

_No response_

### What is the expected behavior?

This is what I'd expect as output
```txt
f688ae26e9cfa3ba6235477831d5122e
ebd396c421067fc6aaad34d1186fbf31
MD4 { byteLength: 16,  } MD5 { byteLength: 16,  }
```

### What do you see instead?

```txt
f688ae26e9cfa3ba6235477831d5122e00000000000000000000000000000000
ebd396c421067fc6aaad34d1186fbf3100000000000000000000000000000000
MD4 { byteLength: 32,  } MD5 { byteLength: 32,  }
```

Notice, that the byteLength is 32 in both cases

### Additional information

However in the MD5 and MD4 definition in `bun-types/types.d.ts` the value of `byteLength` is correct, 16.

```ts
  export class MD5 extends CryptoHashInterface<MD5> {
    constructor();

    /**
     * The number of bytes the hash will produce
     */
    static readonly byteLength: 16;
  }
```

### PR DESCRIPTION
**Pull request by zhuzilin**
fix #1041

the `git_sha[0..8]` would cause compile error on linux...

### CODE REVIEW

File: src/__global.zig [hunk 1]
```diff
@@ -14,9 +14,9 @@ else
 pub const package_json_version_with_sha = if (Environment.git_sha.len == 0)
     package_json_version
 else if (Environment.isDebug)
-    std.fmt.comptimePrint("0.1.{d}_debug ({s})", .{ build_id, Environment.git_sha[0..8] })
+    std.fmt.comptimePrint("0.1.{d}_debug ({s})", .{ build_id, Environment.git_sha })
```
Comment (Jarred-Sumner) [inline src/__global.zig]:
```suggestion
    std.fmt.comptimePrint("0.1.{d}_debug ({s})", .{ build_id, Environment.git_sha[0..@minimum(Environment.git_sha.len, 8)] })
```

File: src/__global.zig [hunk 2]
```diff
@@ -14,9 +14,9 @@ else
...
 else
-    std.fmt.comptimePrint("0.1.{d} ({s})", .{ build_id, Environment.git_sha[0..8] });
+    std.fmt.comptimePrint("0.1.{d} ({s})", .{ build_id, Environment.git_sha });
```
Comment (Jarred-Sumner) [inline src/__global.zig]:
```suggestion
    std.fmt.comptimePrint("0.1.{d} ({s})", .{ build_id, Environment.git_sha[0..@minimum(Environment.git_sha.len, 8)] });
```

File: src/__global.zig (hunk 1 above)
Comment (zhuzilin) [reply src/__global.zig]: fixed.

File: src/__global.zig (hunk 2 above)
Comment (zhuzilin) [reply src/__global.zig]: fixed.

**Pull request merged**